crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -1015,6 +961,71 @@ impl TwoQubitWeylDecomposition {
         specialized.global_phase += tr.arg();
         Ok(specialized)
     }
+}
+
+static IPZ: GateArray1Q = [[IM, C_ZERO], [C_ZERO, M_IM]];
+static IPY: GateArray1Q = [[C_ZERO, C_ONE], [C_M_ONE, C_ZERO]];
+static IPX: GateArray1Q = [[C_ZERO, IM], [IM, C_ZERO]];
+
+#[pymethods]
+impl TwoQubitWeylDecomposition {
+    #[staticmethod]
+    fn _from_state(
+        angles: [f64; 4],
+        matrices: [PyReadonlyArray2<Complex64>; 5],
+        specialization: Specialization,
+        default_euler_basis: EulerBasis,
+        calculated_fidelity: f64,
+        requested_fidelity: Option<f64>,
+    ) -> Self {
+        let [a, b, c, global_phase] = angles;
+        Self {
+            a,
+            b,
+            c,
+            global_phase,
+            K1l: matrices[0].as_array().to_owned(),
+            K1r: matrices[1].as_array().to_owned(),
+            K2l: matrices[2].as_array().to_owned(),
+            K2r: matrices[3].as_array().to_owned(),
+            specialization,
+            default_euler_basis,
+            calculated_fidelity,
+            requested_fidelity,
+            unitary_matrix: matrices[4].as_array().to_owned(),
+        }
+    }
+
+    fn __reduce__(&self, py: Python) -> PyResult<Py<PyAny>> {
+        Ok((
+            py.get_type_bound::<Self>().getattr("_from_state")?,
+            (
+                [self.a, self.b, self.c, self.global_phase],
+                [
+                    self.K1l.to_pyarray_bound(py),
+                    self.K1r.to_pyarray_bound(py),
+                    self.K2l.to_pyarray_bound(py),
+                    self.K2r.to_pyarray_bound(py),
+                    self.unitary_matrix.to_pyarray_bound(py),
+                ],
+                self.specialization,
+                self.default_euler_basis,
+                self.calculated_fidelity,
+                self.requested_fidelity,
+            ),
+        )
+            .into_py(py))
+    }
+
+    #[new]
+    #[pyo3(signature=(unitary_matrix, fidelity=DEFAULT_FIDELITY, _specialization=None))]
+    fn new(
```

Comment:
Is there value in following the pattern of naming this `py_new` and naming the Rust-side constructor `new`?

Reply:
The pattern that's used everywhere in accelerate for this was rust is `foo_inner` and python is `foo`. I think we should normalize on a single pattern across all the crates, but maybe do that in a separate standalone PR?